src/AsmResolver/IO/MemoryMappedInputFile.cs

```diff
@@ -21,9 +19,8 @@ public sealed class MemoryMappedInputFile : IInputFile
         public MemoryMappedInputFile(string filePath)
         {
             FilePath = filePath ?? throw new ArgumentNullException(nameof(filePath));
-            _file = MemoryMappedFile.CreateFromFile(filePath);
-            long fileSize = new FileInfo(filePath).Length;
-            _dataSource = new MemoryMappedDataSource(_file.CreateViewAccessor(0, fileSize), (ulong) fileSize);
+            _file = new MemoryMappedFileShim(filePath);
+            _dataSource = new MemoryMappedDataSource(_file);
```

Comment:
Since we are using a raw unmanaged pointer to the data, couldn't we get rid of `MemoryMappedDataSource` alltogether and just use `UnmanagedDataSource` instead? This also removes the hard dependency for `Span<T>` in the `GetSpan` method (as we wouldn't need it anymore).